include/ppx/command_line_parser.h

```diff
@@ -54,100 +54,115 @@ struct StandardOptions
-            if constexpr (std::is_same_v<T, std::string>) {
-                return value;
-            }
-            else if constexpr (std::is_same_v<T, bool>) {
-                return GetValueAsBool(defaultValue);
-            }
-
-            std::stringstream ss{value};
-            T                 val;
-            ss >> val;
-            if (ss.fail()) {
-                return defaultValue;
-            }
-            return val;
+        auto it = allOptions.find(optionName);
+        if (it == allOptions.cend()) {
+            return defaultValue;
         }
+        auto valueStr = it->second.back();
+        return GetParsedOrDefault<T>(valueStr, defaultValue);
+    }
 
-    private:
-        // For boolean options, accept "true" and "false" as well as numeric values.
-        bool GetValueAsBool(bool defaultValue) const
-        {
-            std::stringstream ss{value};
-            bool              val;
-            ss >> val;
-            if (ss.fail()) {
-                ss.clear();
-                ss >> std::boolalpha >> val;
-                if (ss.fail()) {
-                    return defaultValue;
-                }
-            }
-            return val;
+    // Same as above, but intended for list flags that are specified on the command line
+    // with multiple instances of the same flag
+    template <typename T>
+    std::vector<T> GetOptionValueOrDefault(std::string_view optionName, const std::vector<T>& defaultValues) const
+    {
+        auto it = allOptions.find(optionName);
+        if (it == allOptions.cend()) {
+            return defaultValues;
+        }
+        std::vector<T> parsedValues;
+        T              nullValue{};
+        for (size_t i = 0; i < it->second.size(); ++i) {
+            parsedValues.emplace_back(GetParsedOrDefault<T>(it->second.at(i), nullValue));
         }
+        return parsedValues;
+    }
 
-        std::string name;
-        std::string value;
-    };
+    // Same as above, but intended for resolution flags that are specified on command line
+    // with <Width>x<Height>
+    std::pair<int, int> GetOptionValueOrDefault(std::string_view optionName, const std::pair<int, int>& defaultValue) const;
 
-public:
-    const StandardOptions& GetStandardOptions() const
+    // (WILL BE DEPRECATED, USE KNOBS INSTEAD)
+    // Get the parameter value after converting it into the desired integral,
+    // floating-point, or boolean type. If the value fails to be converted,
+    // return the specified default value.
+    template <typename T>
+    T GetExtraOptionValueOrDefault(std::string_view optionName, const T& defaultValue) const
     {
-        return standardOptions;
+        static_assert(std::is_integral_v<T> || std::is_floating_point_v<T> || std::is_same_v<T, std::string>, "GetExtraOptionValueOrDefault must be called with an integral, floating-point, boolean, or std::string type");
```

Comment:
Ah, seems like clang-format leaves this line as-is, surprising, maybe something in our config.